src/CommunityToolkit.Maui/Views/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,238 @@
+﻿namespace CommunityToolkit.Maui.Views;
+
+using System.Security.Cryptography;
+using System.Text;
+using CommunityToolkit.Maui.Core;
+using Microsoft.Extensions.Logging;
+using Microsoft.Maui.Controls;
+
+/// <summary>Gravatar image source.</summary>
+/// <remarks>Note that <see cref="UriImageSource"/> is sealed and can't be used as a parent!</remarks>
+public class GravatarImageSource : ImageSource, IGravatarImageSource
+{
+	/// <summary>The backing store for the <see cref="CacheValidity" /> bindable property.</summary>
+	public static readonly BindableProperty CacheValidityProperty = BindableProperty.Create(nameof(CacheValidity), typeof(TimeSpan), typeof(UriImageSource), TimeSpan.FromDays(1));
+
+	/// <summary>The backing store for the <see cref="CachingEnabled" /> bindable property.</summary>
+	public static readonly BindableProperty CachingEnabledProperty = BindableProperty.Create(nameof(CachingEnabled), typeof(bool), typeof(UriImageSource), true);
+
+	/// <summary>The backing store for the <see cref="Email" /> bindable property.</summary>
+	public static readonly BindableProperty? EmailProperty = BindableProperty.Create(nameof(Email), typeof(string), typeof(GravatarImageSource), defaultValue: GravatarImageSourceDefaults.DefaultEmail, propertyChanged: OnEmailPropertyChanged);
+
+	/// <summary>The backing store for the <see cref="Image" /> bindable property.</summary>
+	public static readonly BindableProperty ImageProperty = BindableProperty.Create(nameof(Image), typeof(DefaultImage), typeof(GravatarImageSource), defaultValue: GravatarImageSourceDefaults.Defaultimage, propertyChanged: OnDefaultImagePropertyChanged);
+
+	int gravatarSize;
+
+	/// <summary>Initializes a new instance of the <see cref="GravatarImageSource"/> class.</summary>
+	public GravatarImageSource()
+	{
+	}
+
+	/// <summary>On property changed.</summary>
+	/// <param name="propertyName">Property name.</param>
+#pragma warning disable CS8625 // Cannot convert null literal to non-nullable reference type. NOTE: This is to match override
+	protected override void OnPropertyChanged(string propertyName = null)
```

Comment:
```suggestion
	protected override void OnPropertyChanged(string? propertyName = null)
```

Please remove pragma warnings

Reply:
Removed